docs/developers/code-style.md

```diff
@@ -220,6 +365,162 @@ where:
 - `when` - stimulus: some action against the system under test.
 - `then` - expectations/assertions.
 
+#### Tests granularity
+
+Unit tests should be as granular as possible. Try to split unit tests into smaller ones until this is impossible to do.
+
+```
+// bad
+@Test
+public void testFooBar() {
+    // when
+    final String foo = service.getFoo();
+    final String bar = service.getBar();
+    
+    // then
+    assertThat(foo).isEqualTo("foo");
+    assertThat(bar).isEqualTo("bar");
+}
+
+// good
+@Test
+public void testFoo() {
+    // when
+    final String foo = service.getFoo();
+    
+    // then
+    assertThat(foo).isEqualTo("foo");
+}
+
+@Test
+public void testBar() {
+    // when
+    final String bar = service.getBar();
+    
+    // then
+    assertThat(bar).isEqualTo("bar");
+}
+```
+
+This also applies to cases where same method is tested with different arguments inside single unit test.
+Note: This represents the replacement we have selected for parameterized testing.
+
```

Comment:
I'm strongly up to use parametrized testing, why do we need to make duplicated tests instead of using the one parametrized?